docs/api/conventions.md

```diff
@@ -91,29 +91,147 @@ There are a few conventions that Remix uses you should be aware of.
-- **Dots in route filenames**: Adding a `.` in a route file will create a nested URL, but not a nested layout. Flat files are flat layouts, nested files are nested layouts. The `.` allows you to create nested URLs without needing to create a bunch of layouts. For example: `app/routes/some.long.url.tsx` will create the URL `/some/long/url`.
-- **`app/routes/index.tsx`**: Routes named "index" will render when the parent layout route's path is matched exactly.
-- **`$param`**: The dollar sign denotes a dynamic segment of the URL. It will be parsed and passed to your loaders and routes.
+Setting up routes in Remix is as simple as creating files in your `app` directory. These are the conventions you should know to understand how routing in Remix works.
 
-  For example: `app/routes/users/$userId.tsx` will match the following URLs: `users/123` and `users/abc` but not `users/123/abc` because that has too many segments. See the <Link to="../routing">routing guide</Link> for more information.
+Please note that you can use either `.jsx` or `.tsx` file extensions depending on whether or not you use TypeScript. We'll stick with `.tsx` in the examples to avoid duplication (and because we ❤️ TypeScript).
 
-  Some CLIs require you to escape the \$ when creating files:
+#### Root Layout Route
 
-  ```bash
-  touch routes/\$params.tsx
-  ```
+```[3]
+app
+| routes
+| root.tsx
+```
+
+This is your root layout, or "root route" (very sorry for those of you who pronounce those words the same way!). It works just like all other routes:
+
+- You can export a [`loader`](#loader), [`action`](#action), [`meta`](#meta), [`headers`](#headers), or [`links`](#links) function
+- You can export an [`ErrorBoundary`](#errorboundary) or [`CatchBoundary`](#catchboundary)
+- Your default export is the layout component that renders the rest of your app in an [`<Outlet />`](https://reactrouter.com/docs/en/v6/api#outlet)
+
+#### Basic Routes
+
+```[3-4]
+app
+| routes
+| - about.tsx
+| - index.tsx
+| root.tsx
+```
+
+Any JavaScript or TypeScript files in the `app/routes/` directory will become routes in your application. The filename maps to the route's URL pathname, except for `index.tsx` which maps to the root pathname.
+
+The default export in this file is the component that is rendered at that route.
+
+#### Nested Routes
+
+```[3-5]
+app
+| routes
+| - blog
+| --- categories.tsx
+| --- index.tsx
+| - about.tsx
+| - index.tsx
+| root.tsx
+```
+
+Folders inside the `app/routes/` directory will create nested routes and URLs in your app. Files named `index.tsx` will render when the parent layout route's path is matched exactly.
+
+#### Layout Routes
+
+```[3-4]
+app
+| routes
+| - blog
+| --- blog.tsx
+| --- categories.tsx
+| --- index.tsx
+| - about.tsx
+| - index.tsx
+| root.tsx
+```
```

Comment:
I find a really great way to format directory structures is what you get from [`tree`](https://formulae.brew.sh/formula/tree). If you want to just copy/paste some of the characters it uses, check [the jokes tutorial](https://remix.run/docs/en/v1/tutorials/jokes#explore-the-project-structure). For example:

```suggestion
app
├── routes
│   ├── blog
│   │   ├── categories.tsx
│   │   └── index.tsx
│   ├── blog.tsx
│   ├── about.tsx
│   └── index.tsx
└── root.tsx
```

(Note also I think the hierarchy was incorrect because the `blog.tsx` looked like a child of the `blog` directory, so I fixed that as well).